apps/explorer/app/contract/[id]/page.tsx

```diff
@@ -24,97 +42,58 @@ export const revalidate = 0
 
 export default async function Page({ params }) {
   const id = params?.id
-
-  // Grab the contract and previous revisions data.
-  const [
-    [contract, contractError],
-    [previousRevisions, previousRevisionsError],
-    [currentTip, currentTipError],
-  ] = await Promise.all([
-    to(getExplored().contractByID({ params: { id } })),
-    to<ExplorerFileContract[]>(
-      getExplored().contractRevisions({ params: { id } })
-    ),
-    to(getExplored().consensusTip()),
-  ])
-
-  if (contractError) throw contractError
-  if (!contract) return notFound()
-
-  if (previousRevisionsError) throw previousRevisionsError
-  if (!previousRevisions)
-    throw new Error('No previousRevisions found in successful request')
-
-  if (currentTipError) throw currentTipError
-  if (!currentTip) throw new Error('No currentTip found in successful request')
+  let contract: ExplorerFileContract | ExplorerV2FileContract
+  let contractRevisions: ExplorerFileContract[] | ExplorerV2FileContract[]
```

Comment:
I think it would be simpler to do the full v1 and v2 and return in their respective if else blocks.